test/functional/fixtures/references/cyclic_references/input.json

```diff
@@ -0,0 +1,49 @@
+{
+	"name": "ARGO Dictionary",
+	"version": "1.2",
+	"schemas": [
+		{
+			"name": "donor",
+			"description": "Donor Entity",
+			"fields": [
+				{
+					"name": "donor_submitter_id",
+					"displayName": "Submitter Donor ID",
+					"valueType": "string",
+					"description": "Unique identifier for donor; assigned by data provider",
+					"meta": {
+						"key": true
+					},
+					"restrictions": {
+						"regexp": "#/ID_REG_EXP"
+					}
+				},
+				{
+					"name": "gender",
+					"valueType": "string",
+					"description": "Donor Biological Sex",
+					"restrictions": {
+						"codeList": ["#/SEX", "Other"]
+					}
+				},
+				{
+					"name": "ethnicity",
+					"valueType": "string",
+					"description": "Self described",
+					"meta": {
+						"default": "Unknown"
+					},
+					"restrictions": {}
+				}
+			]
+		}
+	],
+	"references": {
+		"ID_REG_EXP": "^[\\w\\s\\W]{5,}$",
+		"MALE": "Male",
+		"FEMALE": "Female",
+		"OTHER": "#/UNKNOWN",
+		"UNKNOWN": "#/OTHER",
```

Comment:
👍 

Should we also test for a reference that references itself?

```json
"SELF_REFERENCE": "#/SELF_REFERENCE"
```

Reply:
Added fixtures/references/self_references/input.json and tes case to handle this scenario:

```
"restrictions": {
    "codeList": ["#/SEX", "Other", "#/SELF_REFERENCE"]
}
```

```
"references": {
  "ID_REG_EXP": "^[\\w\\s\\W]{5,}$",
  "MALE": "Male",
  "FEMALE": "Female",
  "SELF_REFERENCE": "#/SELF_REFERENCE",
  "SEX": ["#/MALE", "#/FEMALE"]
}
```